interview-questions.md

```diff
@@ -14,33 +14,33 @@ Researched answer:
 
-Researched answer:
+Researched answer: same as above but fancier.  answer.toFancierExplanationOfSimpleConcept()
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: The AND &&, OR ||, NOT !, greater than >, less than <, greater than or equal to >=, less than or equal to <=, loosely equal to ==, and strictly equal to === operators are logical operators which give you a result of true or false.
 
-Researched answer:
+Researched answer: A boolean is a datatype that has only two values, true or false. This is important to programming because as code becomes more complicated, the returns will always be one of 6 (or 7 if you count BigInt) primitive datatypes. Being able to set conditionals to the result of a boolean is an integral part of programming.
 
 3. What is an index? What is the difference between index and value?
 
-Your answer:
+Your answer: An index is basically the address of an element or item in an array or string, whereas a value is what is located at the address. I like using the physical address example of country, state, county, city, street address. Your street address will tell someone where you live and they value there is your house.
 
-Researched answer:
+Researched answer: An index is also known as "position" of a specified value. Values get a lot more complicated if you don't think of them as plainly "Things". Values all play different roles depending on the type of value it is. They can be numbers, functions, variables, arrays, and etc.
 
 4. What is iteration?
 
-Your answer:
+Your answer: It's basically what turn you're at when you do a repetitive task. When you iterate through something, you keep repeating it as many times as necessary until some condition is met. Plainly speaking, a farm worker will move through his crops every day (iteration) until there are no more crops to pick that day (end condition).
 
-Researched answer:
+Researched answer: Iterations are also known as loops, they allow you to do repetitive asks while allowing you to not reeat yourself. Iteration is very handy when you want to do something over and over again while only adjusting a few pieces of information. 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Pair programming is where two people work together as a driver and navigator, the driver shares his screen and follows the directions of the navigator, who is responsible for directing the driver, watching for errors, and researching topics. The driver and navigator switch positions regularly so that both are engaged, which both increases productivity and efficiency, as the pair work together to solve problems.
 
-Researched answer:
+Researched answer: The concept of Pair Programming comes from Agile Software Development techniques.toFancierExplanationOfSimpleConcept()
 
 ## Looking Ahead: Terms for Next Week
 
```

Comment:
It's always great to look ahead at the terms so you have some context when hearing the terms during lectures. On the next assessment, make sure to define the terms.